src/library/vm/vm_json.cpp

```diff
@@ -77,7 +77,7 @@ vm_obj to_obj(json const & j) {
         return mk_vm_constructor(json_idx::vfloat, to_obj(f));
     } else if (j.is_number()) {
```

Comment:
I think you need to check `is_number_unsigned` and `is_number_signed` separately

Reply:
The code is the same for both cases no?

Reply:
https://nlohmann.github.io/json/features/types/number_handling/  We should also use `uint64_t` and `int64_t` instead of `int`.

Reply:
I had a quick go at using mpz directly in the json parser, but I think heap objects aren't supported

Reply:
@EdAyers: it's not the same, you need a different type for `i` in the two cases